### PR TITLE
Extract errors into errors.js

### PR DESCRIPTION
**Pull request by brandon93s**
- Extract errors into their own file
- Export each error individually 
- Apply each error to `got`, maintaining functionality  

----

Related to #418

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -597,89 +598,10 @@ for (const method of methods) {
...
-			method: opts.method,
-			path: opts.path,
-			protocol: opts.protocol,
-			url: opts.href
-		});
+for (const error in errors) {
+	if (Object.prototype.hasOwnProperty.call(errors, error)) {
+		got[error] = errors[error];
 	}
 }
```
Comment (sindresorhus) [inline index.js]:
```js
for (const error of Object.keys(errors)) {
	got[error] = errors[error];
}
```

**Comment by sindresorhus** [comment]
You need to add `errors.js` to the `files` property in package.json

File: index.js (hunk 1 above)
Comment (sindresorhus) [reply index.js]: Actually, can just use Object.assign:

```js
Object.assign(got, errors);
```

**Comment by panva** [comment]
How would you feel about exposing `StdError` as well? When working with `got` promise rejections a developer might find checking for the following useful.

```js
((async () => {
  try {
    await got.get(url);
  } catch (err) {
    if (err instanceof StdError) {
      //
    } else {
      //
    }
  }
})();

maybe considering a different name, such as GotError too?

**Comment by sindresorhus** [comment]
Yeah, I think we should expose it too and rename it to `GotError`.

File: index.js (hunk 1 above)
Comment (brandon93s) [reply index.js]: > Actually, can just use Object.assign:

Ah, duh!

**Comment by brandon93s** [comment]
- Renamed to `GotError` and exposed
- Simplify with `Object.assign`
- Include `errors.js` in `files`

**Comment by sindresorhus** [comment]
Looks good :)

**Pull request merged**